### PR TITLE
Added create/delete registered model to R API

### PR DESCRIPTION
**Pull request by bramrodenburg**
… Rodenburg <14278376+bramrodenburg@users.noreply.github.com>

## What changes are proposed in this pull request?

Adding basic Model Registry functionality to the R API. This is currently missing in the Python API, and therefore R users have to fall back to the Python API for doing Model Registry related operations. 

In this pull request I've added the functions `mlflow_create_registered_model` and `mlflow_delete_registered_model`. After this PR, my idea is to add the other Model Registry related operations to the R API as well.

## How is this patch tested?

- Added R unit tests

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

Added basic model registry functionality to the MLflow R interface. 

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [X] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [X] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [X] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/R/mlflow/R/model-registry.R [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+#' @export
+mlflow_create_registered_model <- function(name, tags = NULL, description = NULL, client = NULL) {
+  client <- resolve_client(client)
+
+  response <- mlflow_rest(
+    "registered-models",
+    "create",
+    client = client,
+    verb = "POST",
+    version = "2.0/preview",
```
Comment (ankit-db) [inline mlflow/R/mlflow/R/model-registry.R:18]: I'm not sure we need the preview routes here

**Review by ankit-db — COMMENTED** [review]
Seems reasonable! Will let @tomasatdatabricks officially approve

File: mlflow/R/mlflow/R/model-registry.R (hunk 1 above)
Comment (bramrodenburg) [reply mlflow/R/mlflow/R/model-registry.R:18]: I believe all the model registry endpoints still require the `preview` route: https://mlflow.org/docs/latest/rest-api.html#create-registeredmodel

Comment (tomasatdatabricks) [reply mlflow/R/mlflow/R/model-registry.R:18]: I think the documentation is outdated, it should point to the non-preview paths. Thanks for pointing that out @bramrodenburg!

**Review by tomasatdatabricks — CHANGES_REQUESTED** [review]
The PR looks great, thanks for the contribution! Just one request,  can you drop the 'preview' from the used paths? Our documentation is sadly outdated but we should not be using preview paths in new code (the server supports both).

File: mlflow/R/mlflow/R/model-registry.R (hunk 1 above)
Comment (bramrodenburg) [reply mlflow/R/mlflow/R/model-registry.R]: Ah ok! You're welcome :)

Removed the preview routes!

**Review by tomasatdatabricks — APPROVED** [review]
lgtm - thanks for the contribution!

**Pull request merged**